src/components/PoseSetter.tsx

```diff
@@ -0,0 +1,138 @@
+import Pose from './Pose';
+import { Color } from 'three';
+
+/**
+ * A 3D arrow positioned with its tail at (x, y) and its tip pointing in
+ * direction theta. Occupies a 1x1x1 volume.
+ */
+export default function PoseSetter(props: {
+  enabled: boolean;
+  callback: (poseX: number, poseY: number, poseTheta: number) => void;
+}) {
+  const [poseLoc, setPoseLoc] = useState(new THREE.Vector3());
+  const [theta, setTheta] = useState(0);
+  const [dragging, setDragging] = useState(false);
+
+  // Handle mouse location tracking
+  const three = useThree();
+  if (!three)
+    throw new Error(
+      'PoseSetter must be rendered inside a react-three-fiber <Canvas> component.'
+    );
+
+  const pointerMoveEvent = useRef<PointerEvent | null>(null);
+
+  // https://github.com/react-spring/react-three-fiber/blob/889e5734c5e64c951bde6d9b0b042f99449c3c15/src/canvas.tsx#L290
+  // https://github.com/loganzartman/f1tenth-webviz/blob/master/src/index.js#L320
+  const getNdcMousePos = useCallback(() => {
+    const x = pointerMoveEvent.current?.clientX ?? 0;
+    const y = pointerMoveEvent.current?.clientY ?? 0;
+    const { left, right, top, bottom } = three.size;
+    return new THREE.Vector3(
+      ((x - left) / (right - left)) * 2 - 1,
+      -(((y - top) / (bottom - top)) * 2 - 1),
+      0
+    );
+  }, [three.size]);
+
+  const getWorldMousePos = useCallback(() => {
+    return getNdcMousePos().unproject(three.camera);
+  }, [getNdcMousePos, three.camera]);
+
+  // Handle pose setting mouse events
+  const mouseDownHandler = useCallback(() => {
+    if (!props.enabled) {
+      return;
+    }
+
+    setDragging(true);
+    setPoseLoc(getWorldMousePos());
+  }, [getWorldMousePos, props.enabled]);
+
+  const mouseMoveHandler = useCallback(
+    (event: MouseEvent) => {
+      if (!props.enabled) {
+        return;
+      }
+      if (!dragging) return;
+      event.stopPropagation();
+
+      const pos = getWorldMousePos();
+      const dragDx = pos.clone().sub(poseLoc);
+      setTheta(Math.atan2(dragDx.y, dragDx.x));
+    },
+    [dragging, getWorldMousePos, poseLoc, props.enabled]
+  );
+
+  const mouseUpHandler = useCallback(() => {
+    setDragging(false);
+    if (!props.enabled) {
+      return;
+    }
+
+    props.callback(poseLoc.x, poseLoc.y, theta);
+  }, [poseLoc.x, poseLoc.y, props, theta]);
+
+  const pointerMoveHandler = useCallback((event: PointerEvent) => {
+    pointerMoveEvent.current = event;
+  }, []);
+
+  useEffect(
+    function setupMouseHandlers() {
+      document
+        .querySelector('canvas')
+        ?.addEventListener('mousedown', mouseDownHandler);
+      document
+        .querySelector('canvas')
+        ?.addEventListener('mouseup', mouseUpHandler);
+      document
+        .querySelector('canvas')
+        ?.addEventListener('pointermove', pointerMoveHandler);
+      document
+        .querySelector('canvas')
+        ?.addEventListener('mousemove', mouseMoveHandler);
```

Comment:
skim the docs for [pointer events](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events), but TLDR is you should be able to use `pointerdown`, `pointerup`, and `pointermove` in place of mouse events and get free touch+mouse support at the same time.

Since [PointerEvents extend MouseEvent](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events#Interfaces), you should not have to rewrite the handlers and you should be able to combine the pointer move and mouse move handlers into just a pointer move handler (and maybe you don't need the pointer event ref that way)